src/controllers/engine/controllerengine.cpp

```diff
@@ -421,63 +393,76 @@ bool ControllerEngine::execute(QScriptValue functionObject,
     if (m_pEngine == nullptr) {
         return false;
     }
-    QScriptValueList args;
-    args << QScriptValue(channel);
-    args << QScriptValue(control);
-    args << QScriptValue(value);
-    args << QScriptValue(status);
-    args << QScriptValue(group);
+    QJSValueList args;
+    args << QJSValue(channel);
+    args << QJSValue(control);
+    args << QJSValue(value);
+    args << QJSValue(status);
+    args << QJSValue(group);
     return internalExecute(m_pEngine->globalObject(), functionObject, args);
 }
 
-bool ControllerEngine::execute(QScriptValue function, const QByteArray data,
+bool ControllerEngine::execute(QJSValue function, const QByteArray data,
                                mixxx::Duration timestamp) {
     Q_UNUSED(timestamp);
     if (m_pEngine == nullptr) {
         return false;
     }
-    QScriptValueList args;
-    args << m_pBaClass->newInstance(data);
-    args << QScriptValue(data.size());
+    QJSValueList args;
+    args << byteArrayToScriptValue(data);
+    args << QJSValue(data.size());
     return internalExecute(m_pEngine->globalObject(), function, args);
 }
 
-/* -------- ------------------------------------------------------
-   Purpose: Check to see if a script threw an exception
-   Input:   QScriptValue returned from call(scriptFunctionName)
-   Output:  true if there was an exception
-   -------- ------------------------------------------------------ */
-bool ControllerEngine::checkException() {
-    if (m_pEngine == nullptr) {
-        return false;
-    }
+// Check if a script evaluation threw an exception. If so, register that the source
+// file threw and error and show error dialog.
+//
+// Input: QJSValue returned from evaluation
+// Output: true if there was an exception, false otherwise.
+QJSValue ControllerEngine::evaluateProgram(const QString& program, const QString& fileName,
+		int lineNumber) {
+	if (m_pEngine == nullptr) {
+		throw NullEngineException();
+	}
+
+	QJSValue evaluationResult = m_pEngine->evaluate(program, fileName, lineNumber);
+	handleEvaluationException(evaluationResult);
+
+    return evaluationResult;
+}
 
-    if (m_pEngine->hasUncaughtException()) {
-        QScriptValue exception = m_pEngine->uncaughtException();
-        QString errorMessage = exception.toString();
-        QString line = QString::number(m_pEngine->uncaughtExceptionLineNumber());
-        QStringList backtrace = m_pEngine->uncaughtExceptionBacktrace();
-        QString filename = exception.property("fileName").toString();
+void ControllerEngine::handleEvaluationException(QJSValue evaluationResult) {
```

Comment:
I think this function name is a little misleading. How about changing it to "checkForEvaluationException"?